code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/RancherConfigToComposeFormatter.java

```diff
@@ -0,0 +1,7 @@
+package io.cattle.platform.servicediscovery.service.impl;
+
+import io.cattle.platform.servicediscovery.resource.ServiceDiscoveryConfigItem;
+
+public interface RancherConfigToComposeFormatter {
+    public Object format(ServiceDiscoveryConfigItem item, Object valueToTransform);
```

Comment:
Actually I'm taking my words back. Returning the null indicates that the formatter didn't apply, and we should continue looping through the rest of formatters. The first not-null response indicates that the formatter modified the data, and the loop breaks.